### ISSUE TITLE
Existential deposits

### ISSUE THREAD

**Issue by lumir-mrkva** [galacticcouncil/HydraDX-node#207]
currently is set to 0, because we allow users to pay transactions fee with 3rd party currencies. so theoretically u don't have to touch HDX ever.

ideally, we would determine the existence of accounts by the sum of value across all assets and compare it with some constant DAI minimal value.

different solutions should be explored:
- always store this sum and update it with every transfer (could be slow)
- allow off-chain audit of account of balances and reimburse bots removing accounts (possibly abusable)

**Comment by enthusiastmartin** [comment]
Another possible solution to consider would be to use the selected currency in which user wants to pay as currency determining the account existence.  
Because it is not allowed to select currency with no balance. 

But does not deal with destroying account where there are assets for this account. But perhaps it could be used as first step to mark account for "destroy" - so the off chain audit can only go through these marked accounts and not all of them.

Just an idea.

**Comment by jak-pan** [comment]
> Another possible solution to consider would be to use the selected currency in which user wants to pay as currency determining the account existence.
> Because it is not allowed to select currency with no balance.
> 
> But does not deal with destroying account where there are assets for this account. But perhaps it could be used as first step to mark account for "destroy" - so the off chain audit can only go through these marked accounts and not all of them.
> 
> Just an idea.

We need to be careful here as the person can have $1 worth of token he is paying the txs with but hold $10000 in value of other tokens. This would wipe his account clean. I'd go more by the sum of values of these tokens. We can make some "optimizations" to balances / currencies so that user will not be able to do this under normal conditions. 

The problem with the off-chain cleanup is that when a holder that didn't fall below existential deposit before, will fall into this range, it will wipe him so, we probably still need to sum. 

Other solution could be to only do this check when the account has transacted in a block making this much easier to handle.

**Comment by jak-pan** [comment]
Let's be also careful about this https://substrate.dev/docs/en/knowledgebase/advanced/account-info (ref-counting) which might actually be the reason we've locked some accounts.

**Comment by enthusiastmartin** [comment]
Using Basilisk pallets (asset registry and duster ) - Existential deposit can be set up in similar way to what has been done on basilisk.

So once #299  is merged - it'd be possible to set it up.

**Comment by lumir-mrkva** [comment]
should be configured in parachain

**Comment by lumir-mrkva** [comment]
already done
